File: projects/07-timetrack/backend/timetrack/src/main/java/com/victor/timetrack/model/TimeEntry.java
package com.victor.timetrack.model;

import jakarta.persistence.*;
import lombok.*;
import org.hibernate.annotations.ColumnDefault;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

@Getter
@Setter
@EqualsAndHashCode(onlyExplicitlyIncluded = true)
@ToString(exclude = {"user","project"})
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name = "time_entries")
public class TimeEntry {
    @Id
    @GeneratedValue
    @EqualsAndHashCode.Include
    private Long id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "user_id", nullable = false)
    private User user;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "project_id", nullable = false)
    private Project project;

    @Column(nullable = false)
    private LocalDate date;

    @Column(nullable = false, precision = 4, scale = 2)
    private BigDecimal hours;

    @Column(nullable = false)
    private String description;

    @Enumerated(EnumType.STRING)
    @ColumnDefault("'DRAFT'")
    @Column(nullable = false)
    private EntryStatus status = EntryStatus.DRAFT;

    private String rejectionNote;

    @CreationTimestamp
    @Column(nullable = false, updatable = false)
    private LocalDateTime createdAt;

    @UpdateTimestamp
    @Column(nullable = false)
    private LocalDateTime updatedAt;
}

File: projects/07-timetrack/frontend/timetrack/src/app/shared/models/time-entry.ts
export const ENTRY_STATUSES = ['DRAFT', 'SUBMITTED', 'APPROVED', 'REJECTED'] as const;

export type EntryStatus = (typeof ENTRY_STATUSES)[number];

export const ENTRY_STATUS_LABELS: Record<EntryStatus, string> = {
  DRAFT: 'Draft',
  SUBMITTED: 'Submitted',
  APPROVED: 'Approved',
  REJECTED: 'Rejected',
};

export interface TimeEntry {
  id: number;
  userId: number;
  userName: string;
  projectId: number;
  projectName: string;
  date: string;
  hours: number;
  description: string;
  status: EntryStatus;
  rejectionNote: string | null;
}

export interface TimeEntryFilters {
  userId?: number;
  projectId?: number;
  status?: EntryStatus;
  month?: string;
}

export interface CreateTimeEntryRequest {
  projectId: number;
  date: string;
  hours: number;
  description: string;
}

export interface UpdateTimeEntryRequest {
  projectId: number;
  date: string;
  hours: number;
  description: string;
}

export interface RejectRequest {
  rejectionNote: string;
}
